### PR TITLE
add setup_firewalld playbook

### PR DESCRIPTION
**Pull request by HVSharma12**
Add `setup_firewalld` playbook.

### CODE REVIEW

File: examples/ansible/setup_firewalld.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      block:
+        - name: Print message requesting immediate reboot
+          ansible.builtin.debug:
+            msg: >-
+               The 'firewalld' package has been uninstalled via transactional-update to ensure compatibility with the intended containerized workload.
+
+        - name: Request immediate reboot  # noqa no-handler
+          ansible.builtin.fail:
+            msg: "Reboot required"
+
```
Comment (pdostal) [inline examples/ansible/setup_firewalld.yml]:
```suggestion
- name: Check if firewalld is installed locally
  stat:
    path: "rpm -q firewalld"
  register: local_firewalld
- name: Fail if firewalld is installed locally
  ansible.builtin.fail:
    msg: "Firewalld is installed locally. Please remove it before installing this container."
  when: local_firewalld.rc != 0
```

File: examples/ansible/setup_firewalld.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    - name: Deploy firewalld using Podman
+      block:
```
Comment (pdostal) [inline examples/ansible/setup_firewalld.yml]: No need for block over here.

File: examples/ansible/setup_firewalld.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          ansible.builtin.command: >-
+            podman container runlabel run "{{ workload.image }}"
+          register: runlabel_run_result
+          changed_when:
+            - "runlabel_run_result.rc == 0"
+
+        - name: Start firewalld using systemd
+          ansible.builtin.service:
+            name: "{{ workload.name }}"
+            state: started
```
Comment (pdostal) [inline examples/ansible/setup_firewalld.yml]: Those two seem to do the same thing.

**Review by pdostal — CHANGES_REQUESTED** [review]

File: examples/ansible/setup_firewalld.yml (hunk 1 above)
Comment (HVSharma12) [reply examples/ansible/setup_firewalld.yml]: stat module only work with path ie. /usr/sbin/firewall, using ansible.builtin.package_facts.

~~~
"msg": "The conditional check 'local_firewalld.rc != 0' failed. The error was: error while evaluating conditional (local_firewalld.rc != 0): 'dict object' has no attribute 'rc'. 'dict object' has no attribute 'rc'\
~~~

File: examples/ansible/setup_firewalld.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  become: true
+  vars:
+    workload:
+      name: firewalld
+      image: registry.opensuse.org/suse/alp/workloads/tumbleweed_images/suse/alp/workloads/firewalld
+
+  tasks:
+    - name: Gather package facts
+      ansible.builtin.package_facts:
+        manager: "rpm"
```
Comment (pdostal) [inline examples/ansible/setup_firewalld.yml:23]: Was this needed? I see this the first time :smile:

Comment (pdostal) [reply examples/ansible/setup_firewalld.yml:23]: Ah I see the task below. This is cool!

File: examples/ansible/setup_firewalld.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        - "('already exist' not in workload_runlabel_install.stdout)"
+
+    - name: Ensure polkit daemon is restarted (if necessary)
+      ansible.builtin.service:
+        name: polkit
+        state: restarted
+      when:
+        - "'etc/polkit-1/actions/org.fedoraproject.FirewallD1.policy' in workload_runlabel_install.stdout"
+
+    - name: Start firewalld using systemd
```
Comment (pdostal) [inline examples/ansible/setup_firewalld.yml]:
```suggestion
    - name: Start and enable firewalld using systemd
```

**Review by pdostal — APPROVED** [review]
Just nits. Otherwise LGTM 🚀

**Review by brett060102 — APPROVED** [review]
Good to me as well.

**Pull request merged**